File: backend/pawnscan/src/main/java/edu/cit/garbo/pawnscan/features/auth/AuthService.java
package edu.cit.garbo.pawnscan.features.auth;

import edu.cit.garbo.pawnscan.features.auth.dto.ForgotPasswordRequest;
import edu.cit.garbo.pawnscan.features.auth.dto.ResetPasswordRequest;
import edu.cit.garbo.pawnscan.features.auth.dto.AuthResponse;
import edu.cit.garbo.pawnscan.features.auth.dto.GoogleAuthConfigResponse;
import edu.cit.garbo.pawnscan.features.auth.dto.GoogleAuthRequest;
import edu.cit.garbo.pawnscan.features.auth.dto.LoginRequest;
import edu.cit.garbo.pawnscan.features.auth.dto.RegisterRequest;
import edu.cit.garbo.pawnscan.features.auth.dto.UserProfileResponse;
import edu.cit.garbo.pawnscan.features.auth.dto.CompleteProfileRequest;
import edu.cit.garbo.pawnscan.features.auth.dto.UserProfileUpdateRequest;
import edu.cit.garbo.pawnscan.features.auth.dto.VerifyOtpRequest;

public interface AuthService {

    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    AuthResponse authenticateWithGoogle(GoogleAuthRequest request);

    GoogleAuthConfigResponse getGoogleAuthConfig();

    AuthResponse getMe(String email);

    UserProfileResponse getProfile(String email);

    UserProfileResponse updateProfile(String email, UserProfileUpdateRequest request);

    AuthResponse completeProfile(String email, CompleteProfileRequest request);

    AuthResponse verifyOtp(VerifyOtpRequest request);

    void forgotPassword(ForgotPasswordRequest request);

    void resetPassword(ResetPasswordRequest request);
}

File: mobile/app/src/main/java/com/cit/pawnscan/features/auth/api/AuthService.kt
package com.cit.pawnscan.features.auth.api

import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.POST
import retrofit2.http.PUT

interface AuthService {
    @POST("/api/auth/register")
    fun register(@Body request: RegisterRequest): Call<AuthResponse>

    @POST("/api/auth/verify-otp")
    fun verifyOtp(@Body request: VerifyOtpRequest): Call<AuthResponse>

    @POST("/api/auth/login")
    fun login(@Body request: LoginRequest): Call<AuthResponse>

    @GET("/api/auth/google/config")
    fun getGoogleAuthConfig(): Call<GoogleAuthConfigResponse>

    @POST("/api/auth/google")
    fun authenticateWithGoogle(@Body request: GoogleAuthRequest): Call<AuthResponse>

    @PUT("/api/auth/complete-profile")
    fun completeProfile(
        @Header("Authorization") authorization: String,
        @Body request: CompleteProfileRequest
    ): Call<AuthResponse>

    @GET("/api/auth/profile")
    fun getProfile(@Header("Authorization") authorization: String): Call<UserProfileResponse>

    @PUT("/api/auth/profile")
    fun updateProfile(
        @Header("Authorization") authorization: String,
        @Body request: UserProfileUpdateRequest
    ): Call<UserProfileResponse>

    @POST("/api/auth/forgot-password")
    fun forgotPassword(@Body request: ForgotPasswordRequest): Call<MessageResponse>

    @POST("/api/auth/reset-password")
    fun resetPassword(@Body request: ResetPasswordRequest): Call<MessageResponse>
}


